plugins/language_sh.lua

```diff
@@ -1,98 +1,111 @@
-    ["for"]       = "keyword",
-    ["break"]     = "keyword",
-    ["continue"]  = "keyword",
-    ["function"]  = "keyword",
-    ["local"]     = "keyword",
-    ["echo"]      = "keyword",
-    ["return"]    = "keyword",
-    ["exit"]      = "keyword",
-    ["alias"]     = "keyword",
-    ["test"]      = "keyword",
-    ["cd"]        = "keyword",
-    ["declare"]   = "keyword",
-    ["enable"]    = "keyword",
-    ["eval"]      = "keyword",
-    ["exec"]      = "keyword",
-    ["export"]    = "keyword",
-    ["getopts"]   = "keyword",
-    ["hash"]      = "keyword",
-    ["history"]   = "keyword",
-    ["help"]      = "keyword",
-    ["jobs"]      = "keyword",
-    ["kill"]      = "keyword",
-    ["let"]       = "keyword",
-    ["mapfile"]   = "keyword",
-    ["printf"]    = "keyword",
-    ["read"]      = "keyword",
+    ["case"] = "keyword",
+    ["in"] = "keyword",
+    ["esac"] = "keyword",
+    ["if"] = "keyword",
+    ["then"] = "keyword",
+    ["elif"] = "keyword",
+    ["else"] = "keyword",
+    ["fi"] = "keyword",
+    ["while"] = "keyword",
+    ["do"] = "keyword",
+    ["done"] = "keyword",
+    ["for"] = "keyword",
+    ["break"] = "keyword",
+    ["continue"] = "keyword",
+    ["function"] = "keyword",
+    ["local"] = "keyword",
+    ["echo"] = "keyword",
+    ["return"] = "keyword",
+    ["exit"] = "keyword",
+    ["alias"] = "keyword",
+    ["test"] = "keyword",
+    ["cd"] = "keyword",
+    ["declare"] = "keyword",
+    ["enable"] = "keyword",
+    ["eval"] = "keyword",
+    ["exec"] = "keyword",
+    ["export"] = "keyword",
+    ["getopts"] = "keyword",
+    ["hash"] = "keyword",
+    ["history"] = "keyword",
+    ["help"] = "keyword",
+    ["jobs"] = "keyword",
+    ["kill"] = "keyword",
+    ["let"] = "keyword",
+    ["mapfile"] = "keyword",
+    ["printf"] = "keyword",
+    ["read"] = "keyword",
     ["readarray"] = "keyword",
-    ["pwd"]       = "keyword",
-    ["select"]    = "keyword",
-    ["set"]       = "keyword",
-    ["shift"]     = "keyword",
-    ["source"]    = "keyword",
-    ["time"]      = "keyword",
-    ["type"]      = "keyword",
-    ["until"]     = "keyword",
-    ["unalias"]   = "keyword",
-    ["unset"]     = "keyword",
-    ["true"]      = "literal",
-    ["false"]     = "literal"
-  }
+    ["pwd"] = "keyword",
+    ["select"] = "keyword",
+    ["set"] = "keyword",
+    ["shift"] = "keyword",
+    ["source"] = "keyword",
+    ["time"] = "keyword",
+    ["type"] = "keyword",
+    ["until"] = "keyword",
+    ["unalias"] = "keyword",
+    ["unset"] = "keyword",
+    ["true"] = "literal",
+    ["false"] = "literal",
+  },
 }
 
+local function merge_tables(a, b) for _, v in pairs(b) do table.insert(a, v) end end
+
+merge_tables(string_interpolation_syntax.patterns, {
+  { pattern = "%$[%w_]+", type = "keyword2" },
+  { pattern = "%$[@#]", type = "keyword2" },
+  { pattern = "%${.-}", type = "keyword2" },
+  { pattern = { "%$%(", "%)" }, type = "keyword2", syntax = sh_syntax },
```

Comment:
```suggestion
  { pattern = { "%$%(%(", "%)%)" }, type = "keyword2", syntax = sh_syntax },
  { pattern = { "%$%(", "%)" }, type = "keyword2", syntax = sh_syntax },
``` 
Prevents issues with math expressions